apps/spruce/src/pages/image/tabs/BuildInformationTab/BuildInformationNavItem.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState } from "react";
+import { css } from "@emotion/react";
+import { Link } from "react-router-dom";
+import { useImageAnalytics } from "analytics";
+import { SideNavItem } from "components/styles";
+import { ImageTabRoutes, getImageRoute } from "constants/routes";
+import { useTopmostVisibleElement } from "hooks";
+import { tocItems } from "./constants";
+
+type BuildInformationNavItemProps = {
+  currentTab: ImageTabRoutes;
+  imageId: string;
+  scrollContainerId: string;
+};
+
+const BuildInformationNavItem: React.FC<BuildInformationNavItemProps> = ({
+  currentTab,
+  imageId,
+  scrollContainerId,
+}) => {
+  const { sendEvent } = useImageAnalytics();
+  const tabIsActive = ImageTabRoutes.BuildInformation === currentTab;
+
+  const [activeItemId, setActiveItemId] = useState("");
+
+  const elements = Array.from(
+    document.querySelectorAll("h3[id^='toc-item']"),
+  ) as HTMLElement[];
+
+  useTopmostVisibleElement({
+    elements,
+    scrollContainerId,
+    setTopmostVisibleElementId: setActiveItemId,
+  });
```

Comment:
I think this can be refactored and the tracking of `activeItemId` can happen in `useTopmostVisibleElement`
```suggestion

  const elements = Array.from(
    document.querySelectorAll("h3[id^='toc-item']"),
  ) as HTMLElement[];

  const activeItemId = useTopmostVisibleElement({
    elements,
    scrollContainerId,
  });
```